osu.Server.Spectator/Hubs/MultiplayerQueue.cs

```diff
@@ -106,10 +106,12 @@ public async Task AddItem(MultiplayerPlaylistItem item, MultiplayerRoomUser user
         {
             if (dbFactory == null) throw new InvalidOperationException($"Call {nameof(Initialise)} first.");
 
+            bool isNewAddition = item.ID == 0;
```

Comment:
I did this because it leaves osu! backwards compatible with the currently-deployed osu-server-spectator, but I agree this may be confusing.

Let me know if it'd be preferred to add a separate `ChangeItem` path, at the expense of breaking this compatibility.

Reply:
Now separated into two methods.